trulens_eval/trulens_eval/feedback/provider/hugs.py

```diff
@@ -430,6 +430,181 @@ def pii_detection_with_cot_reasons(self, text: str):
 
         return score, reasons
 
+class HallucinationEvaluator:
+    # Assuming the class has initialized API_URL and API_TOKEN
+
+    def __init__(self):
+        self.API_URL = "https://api-inference.huggingface.co/models/vectara/hallucination_evaluation_model"
+        self.API_TOKEN = os.getenv("HF_AUTH_TOKEN")
+        if not self.API_TOKEN:
+            raise ValueError("Please set the HF_AUTH_TOKEN environment variable.")
+        self.headers = {"Authorization": f"Bearer {self.API_TOKEN}"}
+
+    def query(self, payload):
+        response = requests.post(self.API_URL, headers=self.headers, json=payload)
+        return response.json()
+
+    def evaluate_hallucination(self, input1, input2):
```

Comment:
Can we define this as a method under `HuggingFace` class to mirror how other feedback functions are called? Most of the code here will translate over nicely. You can see `positive_sentiment()` as an example for how to define and call the HF API.